Classes/Domain/Resource/Product.php

```diff
@@ -82,8 +83,10 @@ public function extractProperties(array $additional = null): array
     {
         $this->settings = $this->getPluginSettings();
 
+        $processedUri = $this->getProcessedImageUris($this->entity->getListImages());
+
         $resource = [
-            'listImage' => $this->getProcessedImageUri($this->entity->getListImage()),
+            'listImage' => $processedUri,
```

Comment:
I think the key and variable should be plural, because you're actually supplying an array of image URIs. The way to show it would be to use plural.

If you would like backwards compatibility here, you can supply both `listImage` (containing a single URI) as well as the new `listImages`.

```suggestion
            'listImages' => $processedUris,
```